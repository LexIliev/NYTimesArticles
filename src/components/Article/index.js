import React from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Text,
  TouchableOpacity,
  Image
} from 'react-native';
import Icon from 'react-native-vector-icons/Feather';

import styles from './styles';

const Article = ({
  title,
  byline,
  publishedDate,
  media,
  handleOnPress
}) => (
  <View style={styles.container}>
    <View style={styles.wrapper}>
      <View style={styles.articleImageWrapper}>
        <Image
          style={styles.articleImage}
          source={{
            uri: media[0]['media-metadata'][0].url
          }}
        />
        <View style={styles.articleTextWrapper}>
          <Text
            style={[styles.text, styles.articleTitle]}
            numberOfLines={2}
          >
            {title !== '' ? title : ' - '}
          </Text>
          <View style={styles.articleInfoWrapper}>
            <Text style={[styles.text, styles.articleByline]}>
              {byline !== '' ? byline : 'By -'}
            </Text>
            <Text style={[styles.text, styles.articlePublishedDate]}>
              {publishedDate !== '' ? publishedDate : 'N/A'}
            </Text>
          </View>
        </View>
      </View>
      <View style={styles.buttonWrapper}>
        <TouchableOpacity onPress={handleOnPress}>
          <Icon name="chevron-right" size={22} color="#000" />
        </TouchableOpacity>
      </View>
    </View>
  </View>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  byline: PropTypes.string.isRequired,
  publishedDate: PropTypes.string.isRequired,
  media: PropTypes.array.isRequired,
  handleOnPress: PropTypes.func
};

export default Article;
