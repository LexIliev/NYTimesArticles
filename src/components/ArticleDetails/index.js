import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, Image } from 'react-native';

import styles from './styles';

const ArticleDetails = ({
  title,
  abstract,
  byline,
  published_date,
  media
}) => {
  const articleImage = media[0]['media-metadata'][2];

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.abstract}>{abstract}</Text>
      <Image
        style={[styles.articleDetailsImage, { height: articleImage.height }]}
        source={{
          uri: articleImage.url
        }}
      />
      <View>
        <Text style={styles.text}>{byline}</Text>
        <Text style={styles.text}>{published_date}</Text>
      </View>
    </View>
  );
};

ArticleDetails.propTypes = {
  title: PropTypes.string.isRequired,
  abstract: PropTypes.string.isRequired,
  byline: PropTypes.string.isRequired,
  published_date: PropTypes.string.isRequired,
  media: PropTypes.array.isRequired
};

export default ArticleDetails;
