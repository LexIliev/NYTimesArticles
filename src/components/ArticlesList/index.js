import React from 'react';
import PropTypes from 'prop-types';
import { View, FlatList } from 'react-native';

import { Article } from '../index';

const ArticlesList = props => (
  <View>
    <FlatList
      data={props.data}
      showsVerticalScrollIndicator={false}
      // onRefresh={() => {}}
      // refreshing={}
      renderItem={({ item }) => (
        <Article
          key={item.id}
          title={item.title}
          byline={item.byline}
          publishedDate={item.published_date}
          media={item.media}
          handleOnPress={
            () => props.navigation.navigate('Details', { id: item.id })
          }
        />
      )}
      keyExtractor={item => item.id.toString()}
    />
  </View>
);

ArticlesList.propTypes = {
  data: PropTypes.array.isRequired
};

export default ArticlesList;
