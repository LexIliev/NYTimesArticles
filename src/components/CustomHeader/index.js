import React from 'react';
import { Header } from 'react-navigation';
import { View } from 'react-native';

import styles from './styles';

const CustomHeader = props => (
  <View style={styles.container}>
    <Header {...props} />
  </View>
);

export default CustomHeader;
