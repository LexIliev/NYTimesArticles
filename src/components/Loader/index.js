import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, ActivityIndicator } from 'react-native';

import styles from './styles';

const Loader = ({ loaderText }) => (
  <View style={styles.container}>
    <ActivityIndicator size="large" color="#7ff2ce" />
    <Text style={styles.text}>{loaderText}</Text>
  </View>
);

Loader.propTypes = {
  loaderText: PropTypes.string
};

export default Loader;
