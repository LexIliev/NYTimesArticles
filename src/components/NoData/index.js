import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import Icon from 'react-native-vector-icons/Feather';
import PTRView from 'react-native-pull-to-refresh';

import styles from './styles';

const NoData = ({ handleRefresh }) => (
  <PTRView onRefresh={handleRefresh}>
    <View style={styles.container}>
      <Icon name="cloud-off" size={82} color="#7ff2ce" />
      <Text style={styles.text}>
        {'Cannot retrieve data.\n Please, try again later.'}
      </Text>
    </View>
  </PTRView>
);

NoData.propTypes = {
  handleRefresh: PropTypes.func
};

export default NoData;
