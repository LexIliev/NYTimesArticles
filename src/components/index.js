import Article from './Article';
import ArticleDetails from './ArticleDetails';
import ArticlesList from './ArticlesList';
import CustomHeader from './CustomHeader';
import Loader from './Loader';
import NoData from './NoData';

export {
  Article,
  ArticleDetails,
  ArticlesList,
  CustomHeader,
  Loader,
  NoData
};
