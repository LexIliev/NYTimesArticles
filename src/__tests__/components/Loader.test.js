import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Loader } from '../../components';

configure({ adapter: new Adapter() });

describe('Loader component', () => {
  it('should match snapshot and render correctly', () => {
    const component = shallow(<Loader />);

    expect(component).toMatchSnapshot('Primary snapshot');
  });

  it('should render with loaderText prop', () => {
    const component = shallow(<Loader loaderText="Page is loading..." />);
    expect(component).toMatchSnapshot('Secondary button snapshot');
  });
});
