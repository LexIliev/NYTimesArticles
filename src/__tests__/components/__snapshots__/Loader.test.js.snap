// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should match snapshot and render correctly: Primary snapshot 1`] = `ShallowWrapper {}`;

exports[`Loader component should render with loaderText prop: Secondary button snapshot 1`] = `ShallowWrapper {}`;
